src/components/commons/LoginHeader.tsx

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable sonarjs/cognitive-complexity */
+import React from "react";
+import { useLocation, useNavigate } from "react-router-dom";
+import { Logout } from "@mui/icons-material";
+import { useTranslation } from "react-i18next";
+import { HeaderAccount, JwtUser, RootLinkType } from "@pagopa/mui-italia";
+import ReCAPTCHA from "react-google-recaptcha";
+import { Box } from "@mui/material";
+import { proceedToLogin } from "../../utils/api/helper";
+import { ErrorsType } from "../../utils/errors/checkErrorsModel";
+import { onBrowserUnload } from "../../utils/eventListeners";
+import ErrorModal from "../../components/modals/ErrorModal";
+import CheckoutLoader from "../../components/PageContent/CheckoutLoader";
+import { CheckoutRoutes } from "../../routes/models/routeModel";
+import {
+  clearSessionItem,
+  getReCaptchaKey,
+  getSessionItem,
+  SessionItems,
+} from "../../utils/storage/sessionStorage";
+
+export default function LoginHeader() {
+  const { t } = useTranslation();
+  const location = useLocation();
+  const navigate = useNavigate();
+  const currentPath = location.pathname.split("/").slice(-1)[0];
+  const pagoPALink: RootLinkType = {
+    label: "PagoPA S.p.A.",
+    href: "https://www.pagopa.it",
+    ariaLabel: t("mainPage.footer.pagoPA"),
+    title: t("mainPage.footer.pagoPA"),
+  };
+  const loginRoutes: Array<string> = [
+    CheckoutRoutes.ROOT,
+    CheckoutRoutes.LEGGI_CODICE_QR,
+    CheckoutRoutes.INSERISCI_DATI_AVVISO,
+    CheckoutRoutes.DATI_PAGAMENTO,
+    CheckoutRoutes.INSERISCI_EMAIL,
+    CheckoutRoutes.SCEGLI_METODO,
+  ];
+  const sessionLoggedUser = getSessionItem(SessionItems.loggedUser) as
+    | JwtUser
+    | undefined;
+  const [loggedUser, setLoggedUser] = React.useState<JwtUser | undefined>(
+    sessionLoggedUser
+  );
+  const ref = React.useRef<ReCAPTCHA>(null);
+  const [loading, setLoading] = React.useState(false);
+  const [error, setError] = React.useState("");
+  const [errorModalOpen, setErrorModalOpen] = React.useState(false);
+
+  const onError = (m: string) => {
```

Comment:
Is this error exclusively dedicated to the URL request phase to proceed with the login? Perhaps the error handling will need to be further explored in a dedicated PR

Reply:
This onError handles the get /login possible error. 
By now it shows a popup. I'm not sure to link it to the login error page displayed in [this figma](https://www.figma.com/design/m37n6digqVmwL0AdotC6kI/branch/KJI6JQbyJ41uILvNldSz56/pagoPA-Checkout?node-id=7756-47333&t=n5TVOo8KNix5pco5-0) simply because it will be more usable to show a popup on the same page where the user try to login, than show an error page and then come back to original page. 
What do you think?
We could simply explore it better in a furthet PR

Reply:
I agree, it should be explored in a PR and a dedicated Jira task for the next sprint.